ext/page_log/palite/palite.cpp

```diff
@@ -1829,22 +2055,22 @@ class PaliteHandle : public WT_PAGE_LOG_HANDLE {
     discard(uint64_t page_id, uint64_t checkpoint_id, WT_PAGE_LOG_DISCARD_ARGS *args)
     {
         storage.simulate_unstable_network();
-        Storage::Transaction txn = storage.begin_transaction();
-        uint64_t lsn = storage.get_next_lsn(txn.conn);
-        storage.discard_page(txn.conn, table_id, page_id, lsn, args);
-        storage.verify_page_chain(txn.conn, table_id, page_id, lsn);
-        storage.commit_transaction(txn);
+
+        const uint64_t lsn = storage.make_next_lsn();
+        storage.discard_page(table_id, page_id, lsn, args);
         args->lsn = lsn;
+
         LOG_DEBUG("Discard page_id={} at lsn={}, backlink_lsn={}, base_lsn={}", page_id, lsn,
           args->backlink_lsn, args->base_lsn);
+
         return 0;
     }
 
     int
     close()
```

Comment:
Unrelated: Can we make it `void close()`? Also in many other places.  

The more general question could be can we switch to exceptions (or even abortions) wherever it is possible on the C++ side. We already have `safe_call()` that could catch everything. I also think that since this is a testing implementation, it could be great to cast all the warnings to errors.

Reply:
This `close` is called from PALI implementation. That's why the return value.